tests/foreman/cli/test_logging.py

```diff
@@ -0,0 +1,254 @@
+
+        return make_repository(options)
+
+    def test_positive_logging_from_foreman_core(self):
+        """Check that GET command to Hosts API is logged and has request ID.
+
+        :id: 0785260d-cb81-4351-a7cb-d7841335e2de
+
+        :expectedresults: line of log with GET has request ID
+
+        :CaseImportance: Medium
+        """
+
+        GET_line_found = False
+        source_log = '/var/log/foreman/production.log'
+        test_logfile = '/var/tmp/logfile_from_foreman_core'
+        with ssh.get_connection() as connection:
+            # get the number of lines in the log before the test
+            line_count_1 = connection.run('wc -l < {0}'.format(source_log),
+                                          output_format='plain')
+            self.assertEqual(line_count_1.return_code, 0, "BASH command error?")
+            # hammer command for this test
+            result = connection.run('hammer host list')
+            self.assertEqual(result.return_code, 0, "BASH command error?")
+            # get the number of lines in the log after the test
+            line_count_2 = connection.run('wc -l < {0}'.format(source_log),
+                                          output_format='plain')
+            self.assertEqual(line_count_2.return_code, 0, "BASH command error?")
+            # get the log lines of interest, put them in test_logfile
+            result = connection.run('sed -n "{0},{1} p" {2} < {2} > {3}'.format(
+                line_count_1.stdout.strip('\n'),
+                line_count_2.stdout.strip('\n'),
+                source_log,
+                test_logfile))
+            self.assertEqual(result.return_code, 0, "BASH command error?")
+        # use same location on remote and local for log file extract
+        ssh.download_file(test_logfile)
+        # search the log file extract for the line with GET to host API
+        with open(test_logfile, "r") as logfile:
+            for line in logfile:
+                if re.search(r'Started GET \"\/api/hosts\?page=1', line):
+                    self.logger.info('Found the line with GET to hosts API')
+                    GET_line_found = True
+                    # Confirm the request ID was logged in the line with GET
+                    match = re.search(r'\[I\|app\|\w{8}\]', line)
+                    assert match, "Request ID not found"
+                    self.logger.info("Request ID found for logging from foreman core")
+                    break
+        assert GET_line_found, "The GET command to list hosts was not found in logs."
+
+    def test_positive_logging_from_foreman_proxy(self):
+        """Check PUT to Smart Proxy API to refresh the features is logged and has request ID.
+
+        :id: 0ecd8406-6cf1-4520-b8b6-8a164a1e60c2
+
+        :expectedresults: line of log with PUT has request ID
+
+        :CaseImportance: Medium
+        """
+
+        PUT_line_found = False
+        source_log = '/var/log/foreman/production.log'
```

Comment:
Hello @vijay8451 , I asked lzap, he said it would be good to verify the UUID I find in the test as it is now is also present in '/var/log/foreman-proxy/proxy.log' because there are two different processes running. I will add an extra step for that. Thanks for spotting that.

Reply:
Hello @vijay8451, test updated to check for presence of same Request ID in proxy.log

Thank you

Reply:
@swadeley looks good to me :+1: